testing/src/test/java/io/stargate/it/http/AuthApiTest.java

```diff
@@ -0,0 +1,181 @@
+            "Failed to create token: Provided username cassandra and/or password are incorrect");
+  }
+
+  @Test
+  public void authTokenGenerate_MissingSecret() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth/token/generate", host),
+            "",
+            HttpStatus.SC_BAD_REQUEST);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription()).isEqualTo("Must provide a body to the request");
+  }
+
+  @Test
+  public void auth() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth", host),
+            objectMapper.writeValueAsString(new Credentials("cassandra", "cassandra")),
+            HttpStatus.SC_CREATED);
+
+    AuthTokenResponse authTokenResponse = objectMapper.readValue(body, AuthTokenResponse.class);
+    assertThat(authTokenResponse.getAuthToken()).isNotNull();
+  }
+
+  @Test
+  public void auth_BadUsername() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth", host),
+            objectMapper.writeValueAsString(new Credentials("bad_username", "cassandra")),
+            HttpStatus.SC_UNAUTHORIZED);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription())
+        .isEqualTo(
+            "Failed to create token: Provided username bad_username and/or password are incorrect");
+  }
+
+  @Test
+  public void auth_BadPassword() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth", host),
+            objectMapper.writeValueAsString(new Credentials("cassandra", "bad_password")),
+            HttpStatus.SC_UNAUTHORIZED);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription())
+        .isEqualTo(
+            "Failed to create token: Provided username cassandra and/or password are incorrect");
+  }
+
+  @Test
+  public void auth_MissingCredentials() throws IOException {
+    String body =
+        RestUtils.post("", String.format("%s:8081/v1/auth", host), "", HttpStatus.SC_BAD_REQUEST);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription()).isEqualTo("Must provide a body to the request");
+  }
+
+  @Test
+  public void authUsernameToken() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth/usernametoken", host),
+            objectMapper.writeValueAsString(new UsernameCredentials("cassandra")),
+            HttpStatus.SC_CREATED);
+
+    AuthTokenResponse authTokenResponse = objectMapper.readValue(body, AuthTokenResponse.class);
+    assertThat(authTokenResponse.getAuthToken()).isNotNull();
+  }
+
+  @Test
+  public void authUsernameToken_BadUsername() throws IOException {
```

Comment:
do we have a convention for test names?
I saw in other places he `camelCaseConvention`, but here there are mixed with `_`

Reply:
No clue. I've been using `baseTestName_TestCondition`

Reply:
I should probably change my tests to match :) since I'm going through schemaless tests soon I can standardize